packages/plugin-commands-installation/src/recursive.ts

```diff
@@ -307,12 +312,18 @@ export default async function recursive (
 
         const { manifest, writeProjectManifest } = manifestsByPath[rootDir]
         let currentInput = [...params]
+        const ignoredPackages = (manifest.pnpm?.updateConfig?.ignoreDependencies ?? [])
+        if (opts.update && params.length === 0) {
+          currentInput = [...ignoredPackages.map(pkg => `!${pkg}`), ...currentInput]
```

Comment:
why do append currentInput if it will be empty always?

Reply:
I did so because I did not want to make if statement’s condition complex, but this is not strong will so I changed if statement’s condition in this line as you say. [a2dbfaa](https://github.com/pnpm/pnpm/pull/5408/commits/a2dbfaa643cd9dc8efa1257545fd0b9cdfcc3ddc)